### DISCUSSION TITLE
在视频内显示播放进度（当前时间/总时间）

### DISCUSSION THREAD

**Discussion by onionycs** [the1812/Bilibili-Evolved#2618] [功能建议 / Ideas]
![新功能建议](https://user-images.githubusercontent.com/37523020/145223400-44971441-2ceb-40e8-a098-fabd4007b24e.png)
[之前在这](https://github.com/the1812/Bilibili-Evolved/discussions/2592#discussioncomment-1770336)提了，可能没被看到。。
# 有了这个功能后就不用在看视频的时候移动鼠标去看还有多久才能看完
![图片](https://user-images.githubusercontent.com/37523020/145223823-fb4f88e9-1341-4580-95dc-6bd245fbc6e3.png)

**Comment by onionycs** [comment]
# 需求来源

![灵感来源](https://user-images.githubusercontent.com/37523020/145224335-17735268-21ef-4abd-9bf9-24ddca281db8.png)‘

# 在`potplayer`一直都深深的受益这个feature。。

# 我并不知道这件事情的难度，但是叔叔这么厉害:joy::joy:，我就提前🐮🍻了

**Reply by shanchengzi** [reply]
这个扩展能实现左上显示总时长、已播放时长，右上显示时间，中间显示视频标题，我觉得比Potplayer的方案还要稍好些，类似安卓上Mxplayer的方案。不过这个扩展不提供快捷键修改和部分功能的关闭，如果不介意它和你已有的扩展、脚本冲突，可以试试
https://microsoftedge.microsoft.com/addons/detail/%E5%93%94%E5%93%A9%E5%93%94%E5%93%A9bilibili%E6%92%AD%E6%94%BE%E5%99%A8%E6%89%A9%E5%B1%95/kifdlkkkncadbhofogcaggklmgpcipek

**Comment by ChaXxl** [comment]
**其实只要用网页全屏来看视频，底部就有进度条**，挺好用的
![image](https://user-images.githubusercontent.com/40326898/145313346-cd64854f-b235-42bc-a6a8-00197159f2e9.png)

**Reply by shanchengzi** [reply]
台式机或者屏大、分辨率高的笔记本还好，但对屏幕不大的1080P笔记本、平板来说，顶部网页工具栏、底部系统任务栏在网页全屏时牺牲的画面大小就没办法忽视了。如果打开的书签较多、启用侧面书签栏，画面直接变得还不如ipad大。
如果B站能在完整全屏时也给进度条就舒服了，但目前是非全屏时（包括默认大小、宽屏、网页全屏）有细线进度条，在完整的全屏状态下没有，旧版、新版都没有。
全屏时将鼠标一直放到屏幕底部会强制弹出进度条，但那个是很宽的、有一堆按钮的进度条，和非全屏时的细线不同，会挡住一部分画面和字幕。和上面的问题一样，对屏幕不大的笔记本和平板来说，挡的有点难受，对台式机或者大屏高分辨率笔记本来说挡的就可以忽略了。

**Comment by the1812** [comment]
看组件作者 @FoundTheWOUT 有什么想法了(

**Comment by FoundTheWOUT** [comment]
底部不是有进度条吗🤔，我记得可以设置来着

**Reply by shanchengzi** [reply]
底部的进度条只有在非全屏状态下（包括默认大小、宽屏、网页全屏）才会一直显示，是在底部显示一条细线，但在完整的全屏状态下会自动隐藏。
全屏时将鼠标一直放到屏幕底部会强制弹出进度条，但那个是很宽的、有一堆按钮的进度条，和非全屏时的细线不同，会挡住一部分画面和字幕。
如果是台式机或者大屏、高分辨率的笔记本，用网页全屏还成。但对小屏幕的1080P笔记本以及平板来说，网页全屏时画面大小损失就没办法忽略了。特别是对打开网页较多使用了侧面书签栏的来说，网页全屏画面还不如ipad大，想有较好的观感只能进入完整的全屏模式。
其实有一个哔哩哔哩(Bilibili)播放器扩展能实现这个功能，它的时间进度显示不是上面网页截图的那种类Potplayer的同侧堆叠式，而是类似安卓端Mxplayer的左上显示总时长、当前时长，右上显示时间，中间还可以显示标题。可惜这个不提供快捷键修改，其快捷键和已安装的其它扩展脚本有冲突。

**Comment by timongh** [comment]
偶然发现了一个方法：

先按 F11 网页全屏，然后再打开视频的网页全屏。

**Reply by shanchengzi** [reply]
卧槽牛B，按你说的操作真的完整全屏还有进度条了。太帅了！

**Reply by ChaXxl** [reply]
牛逼

**Comment by Geeyun-JY3** [comment]
#3831 提到了个脚本：https://greasyfork.org/zh-CN/scripts/440820
